challenge_find_my_mates/src/find_my_mates.py

```diff
@@ -0,0 +1,338 @@
+
+                    room_entity = self._robot.ed.get_entity(id=ROOM_ID)
+                    if not room_entity.in_volume(detected_person.pose.extractVectorStamped(), 'in'):
+                        detected_person = None
+
+                    for person in self._people:
+                        if person.pose.extractVectorStamped() - detected_person._pose.extractVectorStamped() > \
+                            self._min_dist and self._robot.ed.get_entities(id=STARTING_POINT).\
+                            _pose.extractVectorStamped() - detected_person.pose.extractVectorStamped() > self._min_dist:
+                            self._people[n] = detected_person
+
+                    if len(self._people) == (n+1):
+                        rospy.loginfo(
+                            "I found someone at {}".format(self._people[n].pose.extractVectorStamped(),
+                                                           block=False))
+
+                        self._robot.ed.update_entity(
+                            id="person" + str(n),
+                            frame_stamped=kdl_conversions.FrameStamped(kdl.Frame(person_pos_kdl.vector), "map"),
+                            type="waypoint")
+
+                        n += 1
+                    if len(self._people) > 3:
+                        self._robot.head.close()
+                        return 'Done'
+
+        self._robot.head.close()
+        # rospy.sleep(2.0)
+        return 'Done'
+
+
+class IdentifyPeople(State):
+    """
+    Navigate to all three people and determine their attributes
+    """
+
+    def __init__(self, robot):
+        State.__init__(self, outcomes=['Done', 'Aborted', 'Failed'])
+        self._robot = robot
+
+    def execute(self, userdata=None):
+        entities = self._robot.ed.get_entities()
+        person_entities = [entity for entity in entities if (entity.is_a("waypoint") and entity.id.startswith("person"))]
+        for person in person_entities:
+            NavigateToWaypoint(self._robot, EntityByIdDesignator(self._robot, id=person.id), radius=0.7)
+            self._robot.head.look_at_standing_person()
+            self._robot.speech.speak("Please tell me your name.", block=True)
+
+            try:
+                result = self._robot.hmi.query('What names?', 'T -> ' + ' | '.join(challenge_knowledge.names), 'T',
+                                               timeout=self._time_out)
+                command_recognized = result.sentence
+            except TimeoutException:
+                command_recognized = None
+            if command_recognized == "":
+                self._robot.speech.speak("I could not hear your name, please tell me your name.")
+            elif command_recognized in challenge_knowledge.names:
+                self._robot.speech.speak(
+                    "OK, I understand your name is {}, but I will think of a nice nickname.".format(command_recognized))
+            else:
+                self._robot.speech.speak(
```

Comment:
This is correct code, but very weird behaviour. The robot says than you can only be named one of X.

Reply:
What would your proposed solution be?